addons/hr_evaluation/hr_evaluation.py

```diff
@@ -1,374 +1,456 @@
+        for evl_rec in self:
+            if self.state == 'new' and vals.get('state') == 'done':
+                raise Warning(_("""Sorry ! You cannot drag this card from the "To Start" column to the "Done" column. You have to drag it to the "Appraisal Sent" column first."""))
+            # avoid recursive process
+            if vals.get('state') == 'pending' and not evl_rec._context.get('send_mail_status'):
+                evl_rec.button_sent_appraisal()
+            if vals.get('interview_deadline') and not vals.get('meeting_id'):
+                if datetime.datetime.now().strftime(DF) > vals.get('interview_deadline'):
+                    raise Warning(_("The interview date can not be in the past"))
+                # creating employee meeting and interview date
+                evl_rec.create_update_meeting(vals)
+            if vals.get('apprasial_manager_ids'):
+                # add followers
+                for follower_id in emp_obj.browse(vals['apprasial_manager_ids'][0][2]):
+                    if follower_id.user_id:
+                        evl_rec.message_subscribe_users(user_ids=[follower_id.user_id.id])
+        return super(hr_evaluation, self).write(vals)
+
+    @api.one
+    def update_appraisal_url(self, url, email_to, token):
+        url = urlparse.urlparse(url).path[1:]
+        if token:
+            url = url + '/' + token
+        self.appraisal_url = url
+        self.email_to = email_to
         return True
 
-    def survey_req_done(self, cr, uid, ids, context=None):
-        for id in self.browse(cr, uid, ids, context=context):
-            flag = False
-            wating_id = 0
-            if not id.evaluation_id.id:
-                raise osv.except_osv(_('Warning!'), _("You cannot start evaluation without Appraisal."))
-            records = id.evaluation_id.survey_request_ids
-            for child in records:
-                if child.state == "draft":
-                    wating_id = child.id
-                    continue
-                if child.state != "done":
-                    flag = True
-            if not flag and wating_id:
-                self.survey_req_waiting_answer(cr, uid, [wating_id], context=context)
-        self.write(cr, uid, ids, {'state': 'done'}, context=context)
+    @api.one
+    def create_token(self, email, url, partner_id):
+        """ Create response with token """
+        survey_response_obj = self.env['survey.user_input']
+        token = uuid.uuid4().__str__()
+        survey_response_obj.create({
+            'survey_id': url.id,
+            'deadline': self.date_close,
+            'date_create': datetime.datetime.now().strftime(DF),
+            'type': 'link',
+            'state': 'new',
+            'token': token,
+            'evaluation_id': self.id,
+            'partner_id': partner_id.id if partner_id else None,
+            'email': email})
+        return token
+
+    @api.one
+    def create_receiver_list(self, appraisal_receiver):
```

Comment:
create_receiver_list ? Its not creating any list, it calls message_post to all receivers, change method name :confused: